src/sentry/api/endpoints/request_join_organization.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import absolute_import
+
+import logging
+
+from rest_framework import serializers
+from rest_framework.response import Response
+from django.db import IntegrityError, transaction
+from django.db.models import Q
+
+from sentry import experiments
+from sentry.api.base import Endpoint
+from sentry.api.validators import AllowedEmailField
+from sentry.app import ratelimiter
+from sentry.models import AuthProvider, InviteStatus, Organization, OrganizationMember
+
+logger = logging.getLogger(__name__)
+
+
+class RequestJoinSerializer(serializers.Serializer):
+    email = AllowedEmailField(max_length=75, required=True)
+    orgSlug = serializers.RegexField(r"^[a-z0-9_\-]+$", max_length=50, required=True)
+
+
+class RequestJoinOrganization(Endpoint):
```

Comment:
You might be able to make this endpoint an `OrganizationEndpoint`. That way you'll get an `Organization` model passed directly to the method. Not sure if that'll work ok with the no permissions thing, but probably worth investigating